vignettes/odin_models.Rmd

```diff
@@ -0,0 +1,196 @@
+---
+title: "Fitting a continuous-time model"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Fitting a continuous-time model}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.width = 7,
+  fig.height = 5)
+lang_output <- function(x, lang) {
+  cat(c(sprintf("```%s", lang), x, "```"), sep = "\n")
+}
+r_output <- function(x) lang_output(x, "r")
+in_pkgdown <- identical(Sys.getenv("IN_PKGDOWN"), "true")
+
+```
+
+While mcstate is primarily designed to work with discrete-time stochastic models written in odin.dust, it is often possible to interact continuous-time deterministic models written in odin (subject to a few restrictions detailed in the [Porting from odin vignette](https://mrc-ide.github.io/odin.dust/articles/porting.html)).
+
+Considering a simple SIR model, as defined in the [odin documentation](https://mrc-ide.github.io/odin/articles/discrete.html):
+$$\begin{align*}
+\frac{dS}{dt} &= -\beta \frac{SI}{N} \\
+\frac{dI}{dt} &= \beta \frac{SI}{N} - \gamma I \\
+\frac{dR}{dt} &= \gamma I \\
+\end{align*}$$
+Where $S$ is the number of susceptibles, $I$ is the number of infected and $R$ is the number recovered; the total population size $N = S + I + R$ is constant. $\beta$ is the infection rate, $\gamma$ is the recovery rate.
+
+We can write the equivalent deterministic odin model as follows:
+
+```{r}
+sir_code <- "initial(S) <- N_init - I_init
+             initial(I) <- I_init
+             initial(R) <- 0
+             initial(N) <- N_init
+             
+             deriv(S) <- -beta * S * I / N
+             deriv(I) <- beta * S * I / N - gamma * I
+             deriv(R) <- gamma * I
+             deriv(N) <- 0
+             
+             beta <- user(0.4)
+             gamma <- user(0.2)
+             N_init <- user(1e5)
+             I_init <- user(1)"
```

Comment:
Either use the form:

```
gen_sir <- odin.dust::odin_dust({
  initial(S) <- N_init - I_init
  ...
  I_init <- user(1)
})
```

or put the code in a file ending in .R and show that then compile the model with 

```
sir <- odin.dust::odin_dust("sir.R")
```

I try and discourage people from writing code in strings, it's very error prone.

you might prefer `sir` over `gen_sir`?